DataDog/datadog-cdk-constructs
# aws-cdk packages as peer dependencies

### What does this PR do?
This PR makes the aws-cdk construct libraries needed in the datadog-cdk-constructs to be peer dependencies. This should keep versions of the cdk libraries in sync with customer's cdk applications dependencies. 
For example lets say a customer has a cdk project that uses `aws-cdk/core` v1.65.0 but Datadog's-cdk-constructs require `aws-cdk/core` v1.71.0 and above, well then the package manager should fail the installation of the Datadog-cdk-constructs because it has an unmet peer dependency.


### Testing Guidelines
I built and created a local package of the datadog-cdk-constructs library using `npm build` and `npm pack` this created a zip file of the dd-cdk-constructs (with the updated peer dependencies). From there I copied the cdk zip into my [lambda-cdk-constructs-demo](https://github.com/DataDog/cloud-integrations-lambdas/tree/master/lambda-cdk-constructs-demo) and used the locally developed version of the cdk. From there I ran npm install after deliberately changing the demo's aws-cdk-core to lower than the required peer dependency of the dd-cdk and I got an expected warning, saying the peer dependency was unmet. 
![Screen Shot 2021-02-11 at 2 55 01 PM](https://user-images.githubusercontent.com/49878080/107709049-6b2ebd80-6c79-11eb-93e0-26e6b74f42cb.png)
Sure enough after checking the npm list, it showed Datadog-cdk-constructs had a missing peer dependency and the customer would need to install it themselves.
![Screen Shot 2021-02-11 at 2 58 34 PM](https://user-images.githubusercontent.com/49878080/107709160-a29d6a00-6c79-11eb-9a16-2eb6abc3afde.png)

After changing the aws-cdk-core back to 1.71.0 or above then the error went away. This was the expected behavior and how I confirmed it locally.



### Additional Notes
I learned more about peer dependencies here: 
https://classic.yarnpkg.com/en/docs/dependency-types/
https://www.youtube.com/watch?v=ChDG13gsDnw (Skip to 1:30)
Addresses this ticket: https://datadoghq.atlassian.net/jira/software/projects/SLS/boards/205?selectedIssue=SLS-972


### Types of Changes


### Check all that apply
